secure/storage/github/src/lib.rs

```diff
@@ -79,13 +85,47 @@ impl Client {
         }
     }
 
+    /// Recursively delete all files, which as a by product will delete all folders
+    pub fn delete_directory(&self, path: &str) -> Result<(), Error> {
+        let files = self.get_directory(path)?;
+        for file in files {
+            if file.ends_with('/') {
+                self.delete_directory(&file[..file.len() - 1])?;
+                continue;
+            }
+            self.delete_file(&file)?;
+        }
+        Ok(())
+    }
+
+    /// Retrieve a list of branches, this is effectively a status check on the repository
+    pub fn get_branches(&self) -> Result<Vec<String>, Error> {
+        let url = format!("{}/repos/{}/{}/branches", URL, self.owner, self.repository);
```

Comment:
this "URL/repose/owner/repository" pattern probably should be abstracted

Reply:
so then we would potentially call format twice :O

I already have one for general purpose accesses.